### PR TITLE
Update xml_to_rdf_mapper.py

### PR DESCRIPTION
**Pull request by ZCRNK**

### CODE REVIEW

**Review by matteoLorenzini — CHANGES_REQUESTED** [review]
Please remove all the comments which are not meant to explain the logic of the script.

All the documents about how to run the script should be reported in the README file according to the template in the issue #11.

The code should commented according to the [techSpec document](https://docs.google.com/document/d/19Kn_2fJMy3s2UbR5t00KVsUO6t3JJwyXGEwVEw9NaNk/edit#)

File: xml_to_rdf_mapper.py [hunk 1]
```diff
@@ -1,103 +1,130 @@
 # author: ETH Zurich, gta digital, Zoé Reinke
 # license: please refer to the license.txt file in our git repository (https://github.com/gtadigital/ProfileParser)
```
Comment (matteoLorenzini) [inline xml_to_rdf_mapper.py:2]: Update the licence. The link point to another repo

File: xml_to_rdf_mapper.py [hunk 2]
```diff
@@ -1,103 +1,130 @@
 # author: ETH Zurich, gta digital, Zoé Reinke
 # license: please refer to the license.txt file in our git repository (https://github.com/gtadigital/ProfileParser)
 
+#run the code by calling the transform(sourceXML, sourceX3ML, generatorPolicy, output_path, uuid_xpath) function
```
Comment (matteoLorenzini) [inline xml_to_rdf_mapper.py:4]: Remove this comment. Reduce to 1 line and put the comment in the README file

File: xml_to_rdf_mapper.py [hunk 3]
```diff
@@ -1,103 +1,130 @@
...
+    
+    if path_pt2=="":
+        path_w = path_pt1
+    else:
+        path_w = path_pt1+"/"+path_pt2
+    
+    return path_w
+
+
+#This function adds a new rdf node to the rdf graph.
```
Comment (matteoLorenzini) [inline xml_to_rdf_mapper.py:41]: Too verbose. Max 2 lines

File: xml_to_rdf_mapper.py [hunk 4]
```diff
@@ -114,202 +141,362 @@ def create_node(gen_ancestor_node, gen_node, count, path_2_value, current_subjec
...
-        triple_object= Literal(literal_value)
+    elif (target_entity_type!= "") and (literal_value!= "") and ("text()" not in triple_object_identifier) and (generator_name == "Literal"):
+        triple_object= Literal(literal_value, lang= literal_lang)
         g.add((current_subject, getattr(eval(relation_namespace), relation), triple_object))
         return triple_object
     
     return
 
-def create_label(gen_ancestor_node, label_gen_node, path_2_value, triple_object):
+
```
Comment (matteoLorenzini) [inline xml_to_rdf_mapper.py:201]: Too verbose max 2 lines

File: xml_to_rdf_mapper.py [hunk 5]
```diff
@@ -114,202 +141,362 @@ def create_node(gen_ancestor_node, gen_node, count, path_2_value, current_subjec
...
                 lang_text= label_gen_node.find("./arg[@name='language']").text
         
         #add label to graph
-        g.add((triple_object, rdfs.label, Literal(label_text, lang= lang_text)))
+        if (label_text is not None) and (label_text!=""):
+            g.add((triple_object, rdfs.label, Literal(label_text, lang= lang_text)))
+
+
+
+#This function does the main part of the xml to rdf transformation process.
```
Comment (matteoLorenzini) [inline xml_to_rdf_mapper.py:253]: Too verbose max 2 lines

File: xml_to_rdf_mapper.py [hunk 6]
```diff
@@ -114,202 +141,362 @@ def create_node(gen_ancestor_node, gen_node, count, path_2_value, current_subjec
...
-            #update current_subj and counter
-            current_subj= triple_obj
-            counter+=1
+                
+    #write graph data in output file
+    f.write(g.serialize(format='pretty-xml'))
+    f.close()
+
+
+# This function should be called for creating an rdf graph according to
```
Comment (matteoLorenzini) [inline xml_to_rdf_mapper.py:483]: NEVER use "Should". This information is not useful to describe the logic of the script.

File: xml_to_rdf_mapper.py (hunk 1 above)
Comment (ZCRNK) [reply xml_to_rdf_mapper.py:2]: The XML_to_RDF_mapping repository does not have a license.txt file yet. Should I remove the comment then? Or is it possible to copy/paste the license from another of gta's repositories (e.g. profileParser)?

Comment (matteoLorenzini) [reply xml_to_rdf_mapper.py:2]: you have to create a new license file

**Pull request merged**